### ISSUE TITLE
`extra_unused_lifetimes` doesn't warn for lifetimes declared on `impl` and never used

### ISSUE THREAD

**Issue by carols10cents** [rust-lang/rust-clippy#6437]
I tried this code (simplified from the actual code):

```rust
pub struct Scalar;

impl<'a> std::ops::AddAssign<&Scalar> for &mut Scalar {
    fn add_assign(&mut self, _rhs: &Scalar) {
        unimplemented!();
    }
}

impl<'b> Scalar {
    pub fn something<'c>() -> Self {
        Self
    }
}
```

I expected to see this happen: I expected to get an unused lifetime warning for all the lifetimes in this example: `'a` because it's not used anywhere in the `AddAssign` impl block (and `AddAssign`'s definition doesn't have any lifetimes), `'b` because it's not used anywhere in the `impl Scalar` block, and `'c` because it's not used anywhere in `fn something`.

Instead, this happened: I only see a warning for `'c`:

```
warning: this lifetime isn't used in the function definition
  --> src/lib.rs:10:22
   |
10 |     pub fn something<'c>() -> Self {
   |                      ^^
   |
   = note: `#[warn(clippy::extra_unused_lifetimes)]` on by default
```

### Meta

- `cargo clippy -V`: On the playground: 0.0.212 (2020-12-03 5be3f9f)
- `rustc -Vv`: Stable on the playground: 1.48.0
